engine/src/main/java/org/hibernate/validator/cfg/context/PropertyTarget.java

```diff
@@ -21,15 +21,46 @@
 	 * <p>
 	 * Until this method is called constraints apply on class level. After calling this method constraints
 	 * apply on the specified property with the given access type.
-	 * </p>
 	 * <p>
 	 * A given property may only be configured once.
-	 * </p>
 	 *
 	 * @param property The property on which to apply the following constraints (Java Bean notation).
 	 * @param type The access type (field/property).
 	 *
 	 * @return A creational context representing the selected property.
+	 *
+	 * @deprecated Since 6.1. Planned for removal. Use either {@link PropertyTarget#field(String)} or
+	 * 		{@link PropertyTarget#getter(String)} instead.
 	 */
+	@Deprecated
```

Comment:
@gsmet While I agree that the two new methods are nicer, I'm unsure whether it's worth the API change (hazzle with WildFly etc.).

Reply:
Btw. one advantage of the original form was that it'll also better work with models that don't distinguish between fields/getters (HV-1000). In that case we could think of either accepting null for `type` (or simply ignoring whatever is passed) or add an overload `property(String name)`. So it feels we're limiting ourselves with this change a little bit.

Reply:
> While I agree that the two new methods are nicer, I'm unsure whether it's worth the API change (hazzle with WildFly etc.).

The old method is still there, it's marked as deprecated.

> Btw. one advantage of the original form was that it'll also better work with models that don't distinguish between fields/getters (HV-1000).

The original form was misleading because there is indeed a difference in treatment between fields and getters and they are totally different beasts.

As for HV-1000 (which is one of the ultimate goals of this PR), there's a good chance the programmatic API entry points will be different (mostly because we will have to define the type for cascaded validation) so we will be able to have a `property()`.

If we end up reusing the same API classes, then we will unmark `property()` as deprecated but I still think keeping `getter()` and `field()` makes sense.

Reply:
@gunnarmorling from what we have right now for json and maps, `PropertyTarget` for new programmatic mapping has just this method:
```java
public interface PropertyTarget {
    PropertyConstraintMappingContext property(String property, Class<?> propertyType);
}
```
as we need to pass the type of the property. Otherwise we would have all the info to create `Property`. Passing this type also serves as a type validation which I think is a nice bonus :).

So most likely we will end up adding another method anyway. 

Having those two shortcuts for field and getter is consistent with what we have for XML configuration (that's what we talked with @gsmet about). Hence if there might be issues with deprecation of `property()` maybe we should leave it as is also but keep the two shortcuts ?

Reply:
> The old method is still there, it's marked as deprecated.

Understood, but the goal of deprecation is removal, no? Feel free do it if you think it's worth it, to me it doesn't necessarily appear that way (given all the discussions we had in the past...).

> As for HV-1000 [...] entry points will be different

Mh, I see. I was hoping that wouldn't be the case.

> so we will be able to have a property().

I see.

Reply:
> Understood, but the goal of deprecation is removal, no?

Maybe at some point. But I agree we might just keep the method here forever.

To be honest, it's more to be clear than getters and fields should be different from the get to and to offer similar APIs for programmatic API and XML definition.